gooddata-sdk/gooddata_sdk/compute/service.py

```diff
@@ -34,13 +34,16 @@ def for_exec_def(self, workspace_id: str, exec_def: ExecutionDefinition) -> Exec
             exec_def: execution definition - this prescribes what to calculate, how to place labels and metric values
          into dimensions
         """
-        response = self._actions_api.compute_report(workspace_id, exec_def.as_api_model(), _check_return_type=False)
+        response, _, http_headers = self._actions_api.compute_report(
+            workspace_id, exec_def.as_api_model(), _check_return_type=False
+        )
 
         return Execution(
             api_client=self._api_client,
             workspace_id=workspace_id,
             exec_def=exec_def,
             response=response,
+            cancel_token=http_headers["X-GDC-CANCEL-TOKEN"] if exec_def.is_cancellable else None,
```

Comment:
This fails when `X-GDC-CANCEL-TOKEN` is not included in http_headers and exec_def.is_cancellable is True. Consider using `http_headers.get("X-GDC-CANCEL-TOKEN")` if key is not presented then None is used.